### PR TITLE
doc: add good practices

### PR DESCRIPTION
**Pull request by Eikix**

### CODE REVIEW

**Review by tdelabro — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -17,6 +17,67 @@ The pull requests will be reviewed and eventually merged into the main repo. See
...
+            DirCreation(String, io::Error),
+
+            #[error("failed to write to file '{0}': {1}")]
+            WriteToFile(String, io::Error),
+        }
+    ```
+
+    -   One cool feature is to be able to raise _transparent_ errors:
+
+        > Errors may use error(transparent) to forward the source and Display methods straight through to an underlying error without adding an additional message. This would be appropriate for enums that need an "anything else" variant.
```
[the remark this answers is not in the document]
Comment (Eikix) [reply CONTRIBUTING.md:67]: C'est une citation de thiserror:
image

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -17,6 +17,67 @@ The pull requests will be reviewed and eventually merged into the main repo. See
...
+
+        ```rust
+            #[derive(Error, Debug)]
+            pub enum MyError {
+                #[error(transparent)]
+                    Other(#[from] anyhow::Error),  // source and Display delegate to anyhow::Error
+            }
+        ```
+
+For more, refer to [the Idiomatic Rust repository](https://github.com/mre/idiomatic-rust). It regroups all the good practices and elegant flows that Rust enables us to write. For instance, [error handling is described in this article](https://edgarluque.com/blog/wrapping-errors-in-rust/), written by Edgar Luque.
```
[the remark this answers is not in the document]
Comment (Eikix) [reply CONTRIBUTING.md:77]: Okay cool!

File: CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tdelabro) [reply CONTRIBUTING.md:67]: haha bha t'es parti pour un PR au repo thiserror ;)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -17,6 +17,67 @@ The pull requests will be reviewed and eventually merged into the main repo. See
...
+
+    -   You should avoid using these methods as much as possible as they make the current thread panic. As a general rule, we want to handle error cases.
+    -   There are some rare cases where `expect()` and `unwrap()` suitable. Then, comment your code and share your reasoning with your co-contributors: why did you need and properly use these methods.
+        > Hint: If you’re having trouble remembering how to phrase expect error messages remember to focus on the word “should” as in “env variable should be set by blah” or “the given binary should be available and executable by the current user”.
+        -   Example - a specific environment variable is essential to your code's execution:
+        ```rust
+        let path = std::env::var("IMPORTANT_PATH").expect("env variable `IMPORTANT_PATH` should be set by `wrapper_script.sh`");
+        ```
+
+-   `Result<T, SomeError>` type should not be used as a function's input parameter.
```
[the remark this answers is not in the document]
Comment (Eikix) [reply CONTRIBUTING.md:32]: dacc

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -17,6 +17,67 @@ The pull requests will be reviewed and eventually merged into the main repo. See
...
+
+-   Error handling: Rust enables a comprehensive Error handling flow. One can define structs/enums items that implement the trait Error. Then, any third party library or code higher in the execution flow can catch and respond properly to failed states. - In this project, we use [thiserror](https://crates.io/crates/thiserror/1.0.24) to describe and construct errors. A good example of this is [compile.rs](./src/compile.rs).
```
[the remark this answers is not in the document]
Comment (Eikix) [reply CONTRIBUTING.md:34]: dacc

File: CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Eikix) [reply CONTRIBUTING.md:67]: xd

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -17,6 +17,74 @@ The pull requests will be reviewed and eventually merged into the main repo. See
...
+    > A collection of lints to catch common mistakes and improve your Rust code.
+
+    -   Update your Rust version: run `rustup update`
+    -   Install clippy: run `rustup component add clippy`
+    -   Run clippy and receive some tips: run `cargo clippy`
+
+-   `expect()` and `unwrap()`:
+
+    -   You should avoid using these methods as much as possible as they make the current thread panic. As a general rule, we want to handle error cases.
+    -   There are some rare cases where `expect()` and `unwrap()` suitable. Then, comment your code and share your reasoning with your co-contributors: why did you need and properly use these methods.
```
Comment (tdelabro) [inline CONTRIBUTING.md:33]: where expect() and unwrap() `are` suitable

Comment (tdelabro) [inline CONTRIBUTING.md:33]: why did you need and properly use these methods`?`

**Review by tdelabro — APPROVED** [review]
Two little errors of syntax to fix and then you can merge

File: CONTRIBUTING.md (hunk 5 above)
Comment (Eikix) [reply CONTRIBUTING.md:33]: okkay

Comment (Eikix) [reply CONTRIBUTING.md:33]: okayyy
